### PR TITLE
IIDM ThreeWindingsTransformer RatedU0 modification

### PR DESCRIPTION
**Pull request by marqueslanauja**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**
Allows to modify the `ratedU0` attribute of a threeWindingsTransformer.
As the structural ratio of the three legs must be preserved, the ratedU of the three legs must be adjusted.

To set the `ratedU0` attribute I added a new method, `setRatedU0`, in the threeWindingsTransformer interface.

I feel more comfortable with the previous interface where `ratedU0` can only be set when the transformer is created.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ThreeWindingsTransformer.java [hunk 1]
```diff
@@ -355,6 +355,11 @@ default List<Leg> getLegs() {
      */
     double getRatedU0();
 
+    /**
+     * Set the rated voltage at the fictitious bus.
+     */
+    ThreeWindingsTransformer setRatedU0(double ratedU0);
```
Comment (zamarrenolm) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ThreeWindingsTransformer.java:361]: add default throwing an unsupported operation exception

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ThreeWindingsTransformerImpl.java [hunk 2]
```diff
@@ -359,6 +359,12 @@ public List<Leg> getLegs() {
         return Arrays.asList(leg1, leg2, leg3);
     }
 
+    @Override
+    public ThreeWindingsTransformer setRatedU0(double ratedU0) {
+        this.ratedU0 = ratedU0;
```
Comment (zamarrenolm) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ThreeWindingsTransformerImpl.java:364]: add validation and notification of change, similar to `setRatedU` of each leg

File: iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/TestThreeWindingsTransformerModification.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        assertEquals(structuralRatioLeg3, getStructuralRatio(t3wt.getLeg3(), t3wt.getRatedU0()));
+    }
+
+    private static double getStructuralRatio(ThreeWindingsTransformer.Leg leg, double ratedU0) {
+        return leg.getRatedU() / ratedU0;
+    }
+
+    @Test
+    void testApplyChecks() {
+        //
```
Comment (zamarrenolm) [inline iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/TestThreeWindingsTransformerModification.java:62]: missing comment?

**Review by zamarrenolm — CHANGES_REQUESTED** [review]

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ThreeWindingsTransformer.java (hunk 1 above)
Comment (marqueslanauja) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ThreeWindingsTransformer.java]: Done

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ThreeWindingsTransformerImpl.java (hunk 2 above)
Comment (marqueslanauja) [reply iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ThreeWindingsTransformerImpl.java:366]: Done

File: iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/TestThreeWindingsTransformerModification.java (hunk 3 above)
Comment (marqueslanauja) [reply iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/TestThreeWindingsTransformerModification.java]: Done

**Review by zamarrenolm — APPROVED** [review]

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ThreeWindingsTransformerModification.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * @author Luma Zamarreño {@literal <zamarrenolm at aia.es>}
+ * @author José Antonio Marqués {@literal <marquesja at aia.es>}
+ */
+public class ThreeWindingsTransformerModification extends AbstractNetworkModification {
+
+    private final String transformerId;
+    private final Double ratedU0;
+
+    public ThreeWindingsTransformerModification(String transformerId, Double ratedU0) {
```
Comment (annetill) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ThreeWindingsTransformerModification.java:27]: I am not sure with need a `Double` here.

**Review by annetill — APPROVED** [review]
Thanks a lot, working very well!

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ThreeWindingsTransformerModification.java (hunk 4 above)
Comment (marqueslanauja) [reply iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ThreeWindingsTransformerModification.java]: Better double as only one attribute can be modified. Changed.

**Pull request merged**